lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,881 @@
+pub struct ChunkedGelfDecoderConfig {
+    /// Options for the chunked GELF decoder.
+    #[serde(default)]
+    pub chunked_gelf: ChunkedGelfDecoderOptions,
+}
+
+impl ChunkedGelfDecoderConfig {
+    /// Build the `ChunkedGelfDecoder` from this configuration.
+    pub fn build(&self) -> ChunkedGelfDecoder {
+        ChunkedGelfDecoder::new(
+            self.chunked_gelf.timeout_secs,
+            self.chunked_gelf.pending_messages_limit,
+            self.chunked_gelf.max_chunk_length,
+            self.chunked_gelf.max_message_length,
+        )
+    }
+}
+
+/// Options for building a `ChunkedGelfDecoder`.
+#[configurable_component]
+#[derive(Clone, Debug, Derivative)]
+#[derivative(Default)]
+pub struct ChunkedGelfDecoderOptions {
+    /// The timeout, in seconds, for a message to be fully received. If the timeout is reached, the
+    /// decoder drops all the received chunks of the timed out message.
+    #[serde(default = "default_timeout_secs")]
+    #[derivative(Default(value = "default_timeout_secs()"))]
+    pub timeout_secs: f64,
+
+    /// The maximum number of pending incomplete messages. If this limit is reached, the decoder starts
+    /// dropping chunks of new messages, ensuring the memory usage of the decoder's state is bounded.
+    /// If this option is not set, the decoder does not limit the number of pending messages and the memory usage
+    /// of its messages buffer can grow unbounded. This matches Graylog Server's behavior.
+    #[serde(default, skip_serializing_if = "vector_core::serde::is_default")]
+    pub pending_messages_limit: Option<usize>,
+
+    /// The maximum length of a single GELF chunk, in bytes. Chunks longer than this length will
+    /// be dropped. If this option is not set, the decoder does not limit the length of chunks and
+    /// the per-chunk memory is unbounded.
+    ///
+    /// This limit takes only into account the chunk's payload and the GELF header bytes are excluded from the calculation.
+    #[serde(skip_serializing_if = "vector_core::serde::is_default")]
+    pub max_chunk_length: Option<usize>,
+
+    /// The maximum length of a single GELF message, in bytes. Messages longer than this length will
+    /// be dropped. If this option is not set, the decoder does not limit the length of messages and
+    /// the per-message memory is unbounded.
+    ///
+    /// Note that a message can be composed of multiple chunks and this limit is applied to the whole
+    /// message, not to individual chunks. This length should always be greater than the `max_chunk_length`.
+    /// This option is useful to limit the memory usage of the decoders's chunk buffer.
+    ///
+    /// This limit takes only into account the message's payload and the GELF header bytes are excluded from the calculation.
+    /// The message's payload is the concatenation of all the chunks' payloads.
+    #[serde(skip_serializing_if = "vector_core::serde::is_default")]
+    pub max_message_length: Option<usize>,
```

Comment:
same as above, should we use `None` by default or `65,535 * 128` bytes, which is the max UDP datagram size times the maximum number of chunks.

Both of those defaults would work (theoretically) the same for the UDP socket source.

Reply:
Let's leave the default as `None` to match the other framers. Could we call this just `max_length`, though, also to match the other framers?

Reply:
done in https://github.com/vectordotdev/vector/pull/20859/commits/8fbfeeb8ab6526d56e34f9a30df225a20c008a0d